### ISSUE TITLE
[Bug] Repeatable - know which row triggered the ajax request

### ISSUE THREAD

**Issue by pxpm** [Laravel-Backpack/CRUD#3440]
# Bug report

### What I did

Setup a repeatable with ajax fields using `include_all_form_fields` to manipulate the returned results based on form condition.

### What I expected to happen

When user has for example 2 rows, I expect to have a way to know which of those rows triggered the request to the endpoint.

### What happened

I can't know which row triggered the ajax request.

### What I've already tried to fix it

Nothing yet, but I think we can use the newly added `backpack_field.parse_value` event in repeatable #3439 to add two hidden fields to the form, one with `repeatable_name` and other with `repeatable_row` so in the end we know which `select` triggered the request. Maybe add another event `backpack_field.parse_value_finished` that will remove those fields from the form again. 

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

**Comment by promatik** [comment]
@pxpm does fields inside repeatable fields work with `include_all_form_fields`?
I had the idea the fields aren't being sent to the server on ajax requests.

**Comment by pxpm** [comment]
Indeed @promatik they don't work, but they will :) 
 Please check #3439

**Comment by promatik** [comment]
Perfect @pxpm 😁 
So I think it's better to make this issue depende on merging #3439, right?

**Comment by pxpm** [comment]
Maybe it's better to not merge more functionality in the open PRs. 
I plan we tackle this one in a separate PR that we could merge after merging the others. 
The other already passes the linked element to the event, but we still need to decide how to implement this one, like I suggested or some better way we find in the meanwhile. 

Let me know if we should stick with my idea or tackle this to merge with that PR. 

Best,
Pedro

**Comment by promatik** [comment]
@pxpm I agree, maybe I expressed myself in a wrong way 😅
I think we should wait for https://github.com/Laravel-Backpack/CRUD/pull/3439 to start on this one 👌

**Comment by tabacitu** [comment]
I say let's do this in 4.2 😀

**Comment by pxpm** [comment]
Hey @tabacitu I am not sure we should postpone this to 4.2. It's not a BC, it's atleast an `issue` that makes repeatable fields behave different than the others. And aiming at having the `editable pivot fields` #3410 in 4.1 (along with fixing the relationships) this becomes like a `must` in the field, otherwise it would break in very common scenarios I think (and then it would become a real `bug`).

I plan to merge the other repeatable PR's (rows, dependants etc) and only after tackle this one. And I think we are done with repeatable (atleast we gave it the functionality we support in other crud fields).

Rephrasing, what I think is that if we are going to put `repeatable` in the spotlights, we should make it work 100%, atleast in the scenarios we already are aware of.

Let me know, 
Pedro

**Comment by pxpm** [comment]
There is a PR for this already open. I am going to close this issue. #3439
